ibis/tests/all/test_string.py

```diff
@@ -74,6 +74,24 @@ def test_string_col_is_unicode(backend, alltypes, df):
             id='re_replace_spark',
             marks=pytest.mark.xpass_backends((Clickhouse, Impala, Spark)),
         ),
+        param(
+            lambda t: t.string_col.re_search(r'\d+'),
+            lambda t: t.string_col.str.contains(r'\d+'),
+            id='re_search_spark',
+            marks=pytest.mark.xpass_backends((PySpark,)),
+        ),
+        param(
+            lambda t: t.string_col.re_extract(r'(\d+)', 0),
+            lambda t: t.string_col.str.extract(r'(\d+)', expand=False),
+            id='re_extract_spark',
+            marks=pytest.mark.xpass_backends((PySpark,)),
+        ),
+        param(
+            lambda t: t.string_col.re_replace(r'\d+', 'a'),
+            lambda t: t.string_col.str.replace(r'\d+', 'a'),
+            id='re_replace_spark',
+            marks=pytest.mark.xpass_backends((PySpark,)),
```

Comment:
More than just `PySpark` will pass here, which is why you're seeing errors in the CI like

```
[XPASS(strict)] Pandas does not pass this test
```

That means that the pandas backend is not failing the test. It's expected to fail because you marked this test as xpassing on `PySpark` only. The `pyspark.mark.xpass_backends` markers implies that any backend not specified in the first argument list should xfail.

The solution here is to figure out which ones _don't_ pass and list those in `pytest.mark.xfail_backends`.